### ISSUE TITLE
Missing values in get_correlation

### ISSUE THREAD

**Issue by andrewheiss** [moderndive/moderndive#32]
Right now, if there is missing data in either the RHS or LHS of the formula in `get_correlation()`, R returns NA. 

``` r
library(tidyverse)
library(moderndive)

df <- data_frame(thing1 = c(1:9, NA),
                 thing2 = rnorm(10))

df %>% 
  get_correlation(thing2 ~ thing1)
#> # A tibble: 1 x 1
#>   correlation
#>         <dbl>
#> 1          NA
```

This is because within `get_correlation()`, `cor()` is called without additional arguments like `use = "complete.obs"`

A few of my students have been confused by this when they use the function on their own data (fortunately all the data in ModernDive is NA-free). I've made them filter out missing values first:

```r
df %>% 
  filter(!is.na(thing1), !is.na(thing2)) %>% 
  get_correlation(thing2 ~ thing1)
#> # A tibble: 1 x 1
#>   correlation
#>         <dbl>
#> 1       0.156
```

It would be fairly easy to include a `...` argument to `get_correlation()` that would feed extra arguments to the main `cor()` function inside. That way it'd be possible to do something like `df %>% get_correlation(y ~ x, use = "complete.obs")`

However, this adds additional complexity to what was probably designed to be a simple wrapper function. I'm not sure what the best solution is for this though. I see at least four possible options, each with pros and cons:

- Add a dots argument to `get_correlation()`
    - Pros: gives `get_correlation()` the full power of `cor()`
    - Cons: too much complexity
- Add a new argument to `get_correlation()` called `na.rm` or something. If it's set to true, run `cor()` with `use = "complete.obs"`; if it's false, run `cor()` like normal
    - Pros: is simpler
    - Cons: uses nonstandard syntax
- Make students filter out missing data first with `filter(!is.na(variable))`
    - Pros: students should already be familiar with the syntax from earlier MD chapters + it makes the idea of removing missing values explicity
    - Cons: if #29 is resolved and multiple RHS variables are allowed, thus returning a correlation matrix, students will need to filter each of the variables individually first (i.e. `filter(!is.na(var1), !is.na(var2), !is.na(varn))`
- Make students not use `get_correlation()` when there's missing data and instead use standard dplyr syntax like `df %>% summarize(correlation = cor(y, x, use = "complete.obs"))`
    - Pros: fits within the dplyr paradigm and shows how to calculate correlation without the MD package
    - Cons: doesn't use the cool wrapper function anymore

**Comment by rudeboybert** [comment]
Thanks for this @andrewheiss. We also have #29 about `get_correlation()` where we would like to allow for more than one variable on the RHS of the `~`. We'll try to get to this during the off-semester season.

**Comment by rudeboybert** [comment]
getting around to this nearly a year later!

@ismayc at this point, I'm 💯about the path of least resistance. Maybe a combo of options 1 & 2?
1. Add `...` argument to `get_correlation()` that gets passed to `stats::cor()`
1. Add `na.rm = TRUE`@ismayc argument to `get_correlation()` that will trigger `cor(use = "complete.obs")`

**Comment by ismayc** [comment]
I like doing both 1 and 2

**Comment by rudeboybert** [comment]
Fixed in #57
